src/commonMain/kotlin/kotools/types/number/StrictlyPositiveInt.kt

```diff
@@ -20,14 +20,83 @@ import kotlin.jvm.JvmInline
  */
 @SinceKotoolsTypes("4.1")
 public fun Number.toStrictlyPositiveInt(): Result<StrictlyPositiveInt> =
-    StrictlyPositiveInt of toInt()
+    runCatching {
+        val value: Int = toInt()
+        StrictlyPositiveInt(value)
+    }
+
+/**
+ * Returns this number as a [StrictlyPositiveInt], which may involve rounding
+ * or truncation, or returns `null` if this number is negative.
+ *
+ * Here's some usage examples:
+ *
+ * ```kotlin
+ * var result: StrictlyPositiveInt? = 1.toStrictlyPositiveIntOrNull()
+ * println(result) // 1
+ *
+ * result = 0.toStrictlyPositiveIntOrNull()
+ * println(result) // null
+ *
+ * result = (-1).toStrictlyPositiveIntOrNull()
+ * println(result) // null
+ * ```
+ *
+ * You can use the [toStrictlyPositiveIntOrThrow] function for throwing an
+ * [IllegalArgumentException] instead of returning `null` when this number is
+ * negative.
+ */
+@ExperimentalNumberApi
+@ExperimentalSinceKotoolsTypes("4.3")
+public fun Number.toStrictlyPositiveIntOrNull(): StrictlyPositiveInt? {
+    val value: Int = toInt()
+    return if (value <= 0) null else StrictlyPositiveInt(value)
+}
+
+/**
+ * Returns this number as a [StrictlyPositiveInt], which may involve rounding
+ * or truncation, or throws [IllegalArgumentException] if this number is
+ * negative.
+ *
+ * Here's some usage examples:
+ *
+ * ```kotlin
+ * val result: StrictlyPositiveInt = 1.toStrictlyPositiveIntOrThrow()
+ * println(result) // 1
+ *
+ * 0.toStrictlyPositiveIntOrThrow() // IllegalArgumentException
+ * (-1).toStrictlyPositiveIntOrThrow() // IllegalArgumentException
+ * ```
+ *
+ * You can use the [toStrictlyPositiveIntOrNull] function for returning `null`
+ * instead of throwing an [IllegalArgumentException] when this number is
+ * negative.
+ */
+@ExperimentalNumberApi
+@ExperimentalSinceKotoolsTypes("4.3")
```

Comment:
The version should be `4.4`.

```suggestion
@ExperimentalSinceKotoolsTypes("4.4")
```